Challenges/GoDownTheRabbitHole/Solutions/RioMH.js

```diff
@@ -0,0 +1,49 @@
+let parentsCache = [];
+
+function countParents(parentId, data) {
+    let count = 1;
+    
+    for (let index = 1; index < data.length; index++) {
+        let item = data[index]
+        if (item.itemId == parentId && item.parentId != null) {
+            count += countParents(item.parentId, data);
```

Comment:
Need that `parentsCache[item.parentId] || countParents(item.parentId, data)` here too.

Another approach would be to start countParents function with:
```
if (parentCache[parentId]) {
  return parentCache[parentId];
}
let count = 1;
...
```
So it still makes a function call. But returns with cached value straight away. You might like this structure more.